src/feature_extraction.py

```diff
@@ -477,6 +477,112 @@ def persist(self, filepath):
         joblib.dump(self.features, filepath)
 
 
+class PreviousApplicationFeatures(BaseTransformer):
+    def __init__(self, **kwargs):
+        self.features = None
+
+    @property
+    def feature_names(self):
+        feature_names = list(self.features.columns)
+        feature_names.remove('SK_ID_CURR')
+        return feature_names
+
+    def fit(self, X, prev_applications, **kwargs):
+        features = pd.DataFrame({'SK_ID_CURR': prev_applications['SK_ID_CURR'].unique()})
+
+        prev_applications_sorted = prev_applications.sort_values(['SK_ID_CURR', 'DAYS_DECISION'])
+        prev_applications_last_3 = prev_applications_sorted.groupby(by=['SK_ID_CURR']).tail(3)
+        prev_applications_last_5 = prev_applications_sorted.groupby(by=['SK_ID_CURR']).tail(5)
+
+        group_object = prev_applications_sorted.groupby(by=['SK_ID_CURR'])['SK_ID_PREV'].count().reset_index()
+        group_object.rename(index=str,
+                            columns={'SK_ID_PREV': 'previous_application_number_of_prev_application'},
```

Comment:
@pknut it looks more like a number of records for ID rather than number of unique previous applications. It could be the same but to ensure that it is go with `.nunique()` rather than `.count()`

Reply:
In the kaggle data description its written: "There is one row for each previous application related to loans in our data sample.". That means in our situation `.count()` is equal to `.nunique()`, although I will change it to `.nunique()` to make the code more understandable.